dash_table/Format.py

```diff
@@ -0,0 +1,200 @@
+from enum import Enum
+import json
+
+class Align(Enum):
+    UNDEFINED = ''
+    RIGHT = '>'
+    LEFT = '<'
+    CENTER = '^'
+    RIGHT_SIGN = '='
+
+class GroupSeparator(Enum):
+    NO = ''
+    YES = ','
+
+class Padding(Enum):
+    NO = ''
+    YES = '0'
+
+class Prefix(Enum):
+    YOCTO = 10**-24
+    ZEPTO = 10**-21
+    ATTO = 10**-18
+    FEMTO = 10**-15
+    PICO = 10**-12
+    NANO = 10**-9
+    MICRO = 10**-6
+    MILLI = 10**-3
+    UNDEFINED = None
+    KILO = 10**3
+    MEGA = 10**6
+    GIGA = 10**9
+    TERA = 10**12
+    PETA = 10**15
+    EXA = 10**18
+    ZETTA = 10**21
+    YOTTA = 10**24
+
+class Sign(Enum):
+    UNDEFINED  = ''
+    NEGATIVE = '-'
+    POSITIVE = '+'
+    PARANTHESES = '('
+    SPACE = ' '
+
+class Symbol(Enum):
+    UNDEFINED = ''
+    CURRENCY = '$'
+    BINARY = '#b'
+    OCTAL = '#o'
+    HEX = '#x'
+
+class Trim(Enum):
+    NO = ''
+    YES = '~'
+
+class Type(Enum):
+    UNDEFINED = ''
+    EXPONENT = 'e'
+    FIXED_POINT = 'f'
+    DECIMAL_OR_EXPONENT = 'g'
+    DECIMAL = 'r'
+    DECIMAL_SI_PREFIX = 's'
+    PERCENTAGE = '%'
+    PERCENTAGE_ROUNDED = 'p'
+    BINARY = 'b'
+    OCTAL = 'o'
+    DECIMAL_INTEGER = 'd'
+    LOWER_CASE_HEX = 'x'
+    UPPER_CASE_HEX = 'X'
+    UNICODE = 'c'
+
+class Format():
+    def __init__(self):
+        self.locale = {}
+        self.prefix = Prefix.UNDEFINED
+        self.specifier = {
+            'align': Align.UNDEFINED,
+            'fill': '',
+            'group': GroupSeparator.NO,
+            'width': '',
+            'padding': Padding.NO,
+            'precision': '',
+            'sign': Sign.UNDEFINED,
+            'symbol': Symbol.UNDEFINED,
+            'trim': Trim.NO,
+            'type': Type.UNDEFINED
+        }
+
+    # Specifier
+    def align(self, align: Align = Align.UNDEFINED):
```

Comment:
I'm afraid we can't use type annotations and support py2.7 If you really want to, there's a comment-based convention https://mypy.readthedocs.io/en/latest/python2.html

Also, if you're already filling in the default below it seems redundant here. Unless you wanted this to function as a setter/getter, in which case you could use `align=None` and `if(align is None): return self.specifier['align']`